system-apps/app-generator/webClient/src/app/app.component.ts

```diff
@@ -0,0 +1,66 @@
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+import { Component, ViewEncapsulation, ViewChild, AfterViewInit, OnInit } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { ZoweApplication } from "./shared/models/application";
+
+@Component({
+	selector: "app-root",
+	encapsulation: ViewEncapsulation.None,
+	templateUrl: "./app.component.html",
+	styleUrls: ["./app.component.scss"]
+})
+export class AppComponent implements AfterViewInit, OnInit {
+	appList: any;
+
+	constructor(private http: HttpClient) {
+    this.appList = [];
+	}
+
+  createCallback() {
+    var t = this;
+    return function(plugin) {
+      t.appList.push(plugin);
+    }
+  }
+
+	ngOnInit() {
+    const MY_PLUGIN_ID = ZoweZLUX.iframe.pluginDef.basePlugin.identifier;
```

Comment:
I get an error on this line in Chrome console: `Cannot read property 'basePlugin' of undefined`. Note: that I had to build app-generator in dev mode to get the meaningful message.

Reply:
This means there is a timing issue. Needs a fix.